src/test/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/PrisonerIepLevelReviewServiceTest.kt

```diff
@@ -354,25 +410,37 @@ class PrisonerIepLevelReviewServiceTest {
 
     @Test
     fun `process TRANSFERRED when offender stays on same incentive level as previous prison`(): Unit = runBlocking {
+      val bookingId = 1234567L
+      val prisonerNumber = "A1244AB"
+
       // Given
-      val prisonOffenderEvent = prisonOffenderEvent("TRANSFERRED")
-      val prisonerAtLocation = prisonerAtLocation(agencyId = "MDI")
-      whenever(prisonApiService.getPrisonerInfo("A1244AB", true)).thenReturn(prisonerAtLocation)
+      val prisonOffenderEvent = prisonOffenderEvent("TRANSFERRED", prisonerNumber)
+      val prisonerAtLocation = prisonerAtLocation(bookingId, prisonerNumber, agencyId = "MDI")
+      whenever(prisonApiService.getPrisonerInfo(prisonerNumber, true)).thenReturn(prisonerAtLocation)
       whenever(prisonApiService.getLocationById(prisonerAtLocation.assignedLivingUnitId, true)).thenReturn(location)
       whenever(prisonApiService.getIepLevelsForPrison(prisonerAtLocation.agencyId, true)).thenReturn(basicStandardEnhanced)
       whenever(prisonApiService.getIepLevels()).thenReturn(globalIncentiveLevels.asFlow())
       val iepDetails = listOf(
         iepDetail(prisonerAtLocation.agencyId, "Standard", "STD", LocalDateTime.now()),
         iepDetail("BXI", "Basic", "BAS", LocalDateTime.now().minusDays(1)),
       )
-      val iepSummary = iepSummary(iepDetails = iepDetails)
+      val iepSummary = iepSummary("Basic", bookingId, iepDetails)
```

Comment:
You'll see that there are a few tweaks, they're not just to improve readability.

The test was failing and after digging I realised that the wiring of the mocks was returning conflicting responses with each other.

e.g. a request would return `bookingId` X and then a separate request would say `bookingId` is Y

This was because of some test data using default values. I made this more explicit and in general pass the desired `bookingId`/`prisonerNumber`.